x-pack/legacy/plugins/reporting/export_types/common/lib/screenshots/open_url.ts

```diff
@@ -16,14 +16,19 @@ export const openUrl = async (
   browser: HeadlessBrowser,
   url: string,
   conditionalHeaders: ConditionalHeaders,
-  logger: LevelLogger
+  logger: LevelLogger,
+  page?: number
 ): Promise<void> => {
+  // If we're moving to another page in the app, we'll want to wait for the app to tell us
+  // it's loaded the next page.
+  const selector = page && page > 1 ? `[data-share-page="${page}"]` : PAGELOAD_SELECTOR;
```

Comment:
This works for now, definitely a sign that we need the screenshot RFC since this logic is getting more brittle and confusing to implement.

Reply:
I know in the RFC, we want to have the selectors passed in through the API. Yeah, having dynamic page load selectors would be hard to account for in that world. I think we can find a way to express the multiple URLs and multiple pageload selectors, but I think we can and should figure it out.

Reply:
Change requested for this line of code - I'd like the logic to be in the `observable.ts` file and just have the `selector` string passed in, instead of having an optional function parameter.

Reply:
Definitely agree. Adding the selector here was definitely more of a last resort